manifests/templates/reconciler-manager-configmap.yaml

```diff
@@ -175,9 +175,6 @@ data:
            - /otelcol-contrib
            args:
            - "--config=/conf/otel-agent-config.yaml"
-           # TODO: Remove this feature gate when opentelemetry semantic conventions are used
-           # in the collector code.
-           - "--feature-gates=-exporter.googlecloud.OTLPDirect"
```

Comment:
Why is this gate removed?

Reply:
The feature gate reverts the Otel Collector to use legacy mode due to [this change](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/main/exporter/collector/breaking-changes.md#breaking-changes-vs-old-googlecloud-exporter) that causes metrics to be matched to `generic_node` type. It was fixed in [this PR](https://github.com/GoogleContainerTools/kpt-config-sync/pull/280) from the shared Otel Collector deployment by correcting the resource mapping, and is no longer necessary for otel-agent.

Reply:
Does this need to be done in this PR? Or can it be done separately?